src/check_config_provider/redis_config_provider.rs

```diff
@@ -23,25 +30,69 @@ impl RedisPartition {
     }
 }
 
+/// The action to perform when receiving a ConfigUpdate
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum ConfigUpdateAction {
+    Add,
+    Delete,
+}
```

Comment:
Oh I was thinking the updates would basically be a "upsert" or "delete".

Reply:
Yeah, upsert would be a better term here, I'll change that